src/utils.jl

```diff
@@ -50,19 +48,21 @@ _adddiag(a::Union{Matrix, SparseMatrixCSC}, v::Real) = _adddiag!(copy(a), v)
 _adddiag(a::Union{Matrix, SparseMatrixCSC}, v::Vector, c::Real) = _adddiag!(copy(a), v, c)
 _adddiag(a::Union{Matrix, SparseMatrixCSC}, v::Vector{T}) where {T<:Real} = _adddiag!(copy(a), v, one(T))
 
+
 function wsumsq(w::AbstractVector, a::AbstractVector)
-    @check_argdims(length(a) == length(w))
+    @check_argdims(eachindex(a) == eachindex(w))
```

Comment:
Maybe we should be less restrictive and just change the loop below instead?

Reply:
Per others, I would like to leave this for a future PR that adds more careful tests to go with such a change